src/resmom/catch_child.c

```diff
@@ -2735,18 +2735,33 @@ send_sisters_deljob_wait(job *pjob)
  * 	rid_job - rid mom of a job that the server says no longer exists
  *
  * @param[in] jobid - char pointer holding jobid
+ * @param[in] msg - message to log if job has been deleted.
  *
  * @return Void
  *
  */
 void
-rid_job(char *jobid)
+rid_job(char *jobid, char *msg)
```

Comment:
I'd rather not pass in a message that is only used for printing.  I'd rather the function return a value that says if it deleted the job or not.  This way the caller can use that to print or not.

Reply:
@bhroam Thanks for your suggestion. But I feel the code is fine as is. The function is called only once, and I just try to match how it was originally invoked.  The original invocation didn't really care whether the job got deleted or not, but strangely just assumed it did, and the message was always printed. If someone else wants to use the function, and would like something done besides logging a message when job is deleted, then the function can be improved...

Reply:
Please let us know if you feel strongly about this @bhroam.

Reply:
If the function is very scenario-specific and is called nowhere else then is there really a need to pass in a message? function can log it by itself, right?

Reply:
@arungrover : The original code in src/resmom/mom_server.c:is_request() under case BAD_OBIT:

log_event(..., ""Job removed, Server rejected Obit")
...
rid(jobid)

Now if I put inside rid(jobid) the "Job removed, Server reject Obit" hardwired message, it would look odd.

Reply:
Thanks Al! I got it.
IMHO (not suggesting you change), there is not much of a need for this function itself because it is small and there is nowhere else we call it from. 
I understand that you want to keep the function generic by passing in the message but I feel the function is already very specific to the case when no background hook is running for the job, it is not in prerun state.

Reply:
I'm not generally a fan of increasing the size of a function (is_request) that is already hundreds of lines long, but we're only talking a couple lines. @arungrover is correct in that this is the only place rid_job() gets called, so it doesn't make a lot of sense to keep it a function. Especially now that we are specializing the conditions for which the job gets deleted. We also get to remove the external definition from the header file. While there is nothing wrong with the changes you propose @bayucan, I tend to agree with @arungrover here. That should also address the concerns from @bhroam.

Reply:
@mike0042 @arungrover @bhroam : Ok then, I'll work on updating this fix, removing the rid_job() function and putting its content in line under case BAD_OBIT in is_request().